doc/mk_sdcard_image.sh

```diff
@@ -0,0 +1,436 @@
+#!/bin/bash -e
+
+MOUNT_POINT="/tmp/mntpoint"
+CUR_STEP=1
+
+usage()
+{
+	echo "###############################################################################"
+	echo "SD card image builder script for current development product."
+	echo "###############################################################################"
+	echo "Usage:"
+	echo "`basename "$0"` <-p image-folder> <-d image-file> [-s image-size-gb] [-u dom0|domd|domf|doma]"
+	echo "	-p image-folder	Base daily build folder where artifacts live"
+	echo "	-d image-file	Output image file or physical device"
+	echo "	-s image-size	Optional, image size in GB"
+	echo "  -u domain	Optional, unpack the domain specified"
+
+	exit 1
+}
+
+print_step()
+{
+	local caption=$1
+	echo "###############################################################################"
+	echo "Step $CUR_STEP: $caption"
+	echo "###############################################################################"
+	((CUR_STEP++))
+}
+
+###############################################################################
+# Inflate image
+###############################################################################
+inflate_image()
+{
+	local dev=$1
+	local size_gb=$2
+
+	print_step "Inflate image"
+	echo "DEV -" $dev
+	if  [ -b "$dev" ] ; then
+		echo "Using physical block device $dev"
+		return 0
+	fi
+
+	echo "Inflating image file at $dev of size ${size_gb}GB"
+	return 0
+
+	local inflate=1
+	if [ -e $1 ] ; then
+		echo ""
+		read -r -p "File $dev exists, remove it? [y/N]:" yesno
+		case "$yesno" in
+		[yY])
+			sudo rm -f $dev || exit 1
+		;;
+		*)
+			echo "Reusing existing image file"
+			inflate=0
+		;;
+		esac
+	fi
+	if [[ $inflate == 1 ]] ; then
+		sudo dd if=/dev/zero of=$dev bs=1M count=$(($size_gb*1024)) || exit 1
+	fi
+}
+
+###############################################################################
+# Partition image
+###############################################################################
+partition_image()
+{
+	print_step "Make partitions"
+
+	sudo parted -s $1 mklabel msdos
+
+	sudo parted -s $1 mkpart primary ext4 1MiB 257MiB
+	sudo parted -s $1 mkpart primary ext4 257MiB 2257MiB
```

Comment:
here you have start of the new partition == end of the previous one,
but below there is a 1MiB gap - is this by intention?

Reply:
yes